src/response.rs

```diff
@@ -54,7 +53,7 @@ pub struct Response {
     stream: Stream,
     // If this Response resulted from a redirect, the Response containing
     // that redirect.
-    previous: Option<Arc<Response>>,
+    pub(crate) previous: Vec<String>,
```

Comment:
This comment needs updating. In particular it should document which way the vector goes (first request to last request, or vice versa).

Reply:
`hist.iter()` returns a reverse iterator. I agree this is confusing, I'll change it to return a normal iterator and use `first()` here instead.